File: app/src/main/java/com/shrikantbadwaik/tmdb/view/base/BaseView.java
package com.shrikantbadwaik.tmdb.view.base;

public interface BaseView {
    boolean isDeviceOnline();

    void showDeviceOfflineError();

    void showLoading();

    void hideLoading();

    void showSuccessMessage(String success);

    void showErrorMessage(String error);

    void showSocketException();

    void showIOException();
}

File: app/src/main/java/com/shrikantbadwaik/tmdb/view/base/BaseView.kt
package com.shrikantbadwaik.tmdb.view.base

interface BaseView {
    fun isDeviceOnline(): Boolean
    fun showDeviceOfflineError()
    fun showSuccessMessage(message: String)
    fun showErrorMessage(message: String)
    fun showSocketException()
    fun showIOException()
}